oauthproxy_test.go

```diff
@@ -581,7 +581,7 @@ func TestPassGroupsHeadersWithGroups(t *testing.T) {
 	const emailAddress = "john.doe@example.com"
 	const userName = "9fcab5c9b889a557"
 
-	groups := []string{"a", "b"}
```

Comment:
I don't think this is correct, the `groups` in the session will still be a list. This is cheating and is testing that a singleton group with the name `a,b` is reflected back as `a,b,` in the header.

We need to test that a list of groups here turns into a comma-delimited string in `X-Forwarded-Groups` in the final `assert`.